src/vm/excep.cpp

```diff
@@ -7314,8 +7318,8 @@ AdjustContextForWriteBarrier(
 {
     WRAPPER_NO_CONTRACT;
 
+#ifndef WIN64EXCEPTIONS
 #ifdef _TARGET_X86_
```

Comment:
WIN64EXCEPTIONS is based on Windows `RUNTIME_FUNCTION` descriptors and funclets. It does not have any coupling with libunwind; or with how C++ try/catch works on given platform. The way C++ try/catch works on Unix is actually different from how WIN64EXCEPTIONS work - Linux C++ try/catch is based on landing pads, not funclets.